packages/p2p-v2/src/components/Modals/PaymentMethods/PaymentMethodModal/__tests__/PaymentMethodModal.spec.tsx

```diff
@@ -0,0 +1,61 @@
+import React, { PropsWithChildren } from 'react';
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import PaymentMethodModal from '../PaymentMethodModal';
+
+const wrapper = ({ children }: PropsWithChildren<unknown>) => <div id='v2_modal_root'>{children}</div>;
+
+describe('PaymentMethodModal', () => {
+    it('should render the component correctly', () => {
+        render(
+            <PaymentMethodModal
+                description='Payment Method Modal Description'
+                isModalOpen={true}
+                onConfirm={jest.fn()}
+                onReject={jest.fn()}
+                primaryButtonLabel='Yes'
+                secondaryButtonLabel='Yes, remove'
+                title='Payment Method Modal'
+            />,
+            { wrapper }
+        );
+        expect(screen.getByText('Payment Method Modal')).toBeInTheDocument();
+        expect(screen.getByText('Payment Method Modal Description')).toBeInTheDocument();
+    });
+    it('should handle onclick when the yes, remove button is clicked', () => {
+        const onConfirm = jest.fn();
+        render(
+            <PaymentMethodModal
+                description='Payment Method Modal Description'
+                isModalOpen={true}
+                onConfirm={onConfirm}
+                onReject={jest.fn()}
+                primaryButtonLabel='Yes'
+                secondaryButtonLabel='Yes, remove'
+                title='Payment Method Modal'
+            />,
+            { wrapper }
+        );
+        const confirmButton = screen.getByText('Yes, remove');
```

Comment:
can we get the buttons by getByRole instead, same with below

Reply:
@ameerul-deriv, these buttons have a `span` wrapping the text, selecting like `.getByRole({name: 'Yes'})` is not possible

Reply:
ah man thats an issue with this component 😢 no worries 👍